pkg/k8s/identitybackend/identity.go

```diff
@@ -0,0 +1,433 @@
+	identity := &v2.CiliumIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   id.String(),
+			Labels: sanitizeK8sLabels(key.GetAsMap()),
+		},
+		SecurityLabels: key.GetAsMap(),
+		Status: v2.IdentityStatus{
+			Nodes: map[string]metav1.Time{
+				c.NodeName: metav1.Now(),
+			},
+		},
+	}
+
+	_, err := c.Client.CiliumV2().CiliumIdentities().Create(identity)
+	return err
+}
+
+func (c *crdBackend) AllocateIDIfLocked(ctx context.Context, id idpool.ID, key allocator.AllocatorKey, lock kvstore.KVLocker) error {
+	return c.AllocateID(ctx, id, key)
+}
+
+// JSONPatch structure based on the RFC 6902
+// Note: This mirros pkg/k8s/json_patch.go but using that directly would cause
+// an import loop.
+type JSONPatch struct {
+	OP    string      `json:"op,omitempty"`
+	Path  string      `json:"path,omitempty"`
+	Value interface{} `json:"value"`
+}
+
+// AcquireReference updates the status field of the CRD corresponding to id
+// with this node. This marks that CRD as used by this node, and will stop it
+// being garbage collected.
+// Note: the lock field is not supported with the k8s CRD allocator.
+func (c *crdBackend) AcquireReference(ctx context.Context, id idpool.ID, key allocator.AllocatorKey, lock kvstore.KVLocker) error {
+	identity := c.get(ctx, key)
+	if identity == nil {
+		return fmt.Errorf("identity does not exist")
+	}
+
+	capabilities := k8sversion.Capabilities()
+	identityOps := c.Client.CiliumV2().CiliumIdentities()
+
+	var err error
+	if capabilities.Patch {
+		var patch []byte
+		patch, err = json.Marshal([]JSONPatch{
+			{
+				OP:    "test",
+				Path:  "/status",
+				Value: nil,
+			},
+			{
+				OP:   "add",
+				Path: "/status",
+				Value: v2.IdentityStatus{
+					Nodes: map[string]metav1.Time{
+						c.NodeName: metav1.Now(),
+					},
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = identityOps.Patch(identity.GetName(), k8sTypes.JSONPatchType, patch, "status")
+		if err != nil {
+			patch, err = json.Marshal([]JSONPatch{
+				{
+					OP:    "replace",
+					Path:  "/status/nodes/" + c.NodeName,
+					Value: metav1.Now(),
+				},
+			})
+			if err != nil {
+				return err
+			}
+			_, err = identityOps.Patch(identity.GetName(), k8sTypes.JSONPatchType, patch, "status")
+		}
+
+		if err == nil {
+			return nil
+		}
+		log.WithError(err).Debug("Error patching status. Continuing update via UpdateStatus")
+		/* fall through and attempt UpdateStatus() or Update() */
+	}
+
+	identityCopy := identity.DeepCopy()
+	if identityCopy.Status.Nodes == nil {
+		identityCopy.Status.Nodes = map[string]metav1.Time{
+			c.NodeName: metav1.Now(),
+		}
+	} else {
+		identityCopy.Status.Nodes[c.NodeName] = metav1.Now()
+	}
+
+	if capabilities.UpdateStatus {
+		_, err = identityOps.UpdateStatus(identityCopy.CiliumIdentity)
+		if err == nil {
+			return nil
+		}
+		log.WithError(err).Debug("Error updating status. This Continuing update via Update")
```

Comment:
This --> Thus ?

Reply:
oops! that was a left over from when that said "This is not fatal"